src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -581,6 +581,23 @@ export class IndexPattern implements IIndexPattern {
 
   async _fetchFields() {
     const fields = await this.fieldsFetcher.fetch(this);
+
+    // preserve user entered displayName
+    if (this.fields) {
+      const displayNameMap = new Map();
+      for (const field of this.fields) {
+        if (field.displayName) {
+          displayNameMap.set(field.name, field.displayName);
+        }
+      }
+
+      for (const field of fields) {
+        if (displayNameMap.has(field.name)) {
+          field.displayName = displayNameMap.get(field.name);
+        }
+      }
+    }
+
```

Comment:
Dear @timroes & @mattkime, theoretically this PR could be ready for review soon, but here's a part that I think we should discuss. If the user  is entering a  `displayName` in a field, it's persisted as part of the field in the index pattern `fields` property.
https://github.com/elastic/kibana/blob/62c70e95dd4d6852b792ac41894437d2a39f0276/x-pack/plugins/lens/public/indexpattern_datasource/types.ts#L12-L12

however, this is overwritten, when the fields are fetched, and therefore this code keeps the old value. 
So I wonder if it would be cleaner to persist the `displayName` values in a separate property like `fieldFormatMap`, 
https://github.com/elastic/kibana/blob/62c70e95dd4d6852b792ac41894437d2a39f0276/x-pack/plugins/lens/public/indexpattern_datasource/types.ts#L15
so adding a new property to the indexPattern called e.g. `displayNameMap` could be an alternative.

Reply:
Yes, I think this is the correct approach

Reply:
the second one, right? using `displayNameMap`

Reply:
@kertal thats what I intended to say but I'm having second thoughts. I'll think it overr and get back to you shortly.

Reply:
Here's my proposal - I don't think we're benefitting from spreading index pattern attributes over different saved object fields. Lets make this a first step toward consolidating. Create a new saved object field called 'attributes' that stores a json string. That json object will have a map named `fields` and the field objects will have an attribute called `displayName`

Reply:
@mattkima continues working on this, have a question. using the savedObjectClient  you retrieve a savedObject that has an attributes property

```TypeScript
const savedObject = await this.savedObjectsClient.get<IndexPatternAttributes>(
      savedObjectType,
      this.id
    );

    const response = {
      version: savedObject.version,
      found: savedObject.version ? true : false,
      title: savedObject.attributes.title,
      timeFieldName: savedObject.attributes.timeFieldName,
      intervalName: savedObject.attributes.intervalName,
```
However persisted it's like this:

```
 "_source" : {
          "index-pattern" : {
            "title" : "kibana_sample_data_logs",
            "timeFieldName" : "timestamp",
            "fields" : """[{"name":"@timestamp","type":"date","esTypes":["date"],"count":0,"scripted":false,"searchable":true,"aggregatable":true,"readFromDocValues":true},{"name":"_id","type":"string","esTypes":["_id"],"count":0,"scripted":false,"searchable":true,"aggregatable":true,"readFromDocValues":false},{"name":"_index","type":"string","esTypes":["_index"],"count":0,"scripted":false,"searchable":tr
```

does it makes sense to create another layer? because if I understood correctly, you suggested creating a `attributes` field on ES saved object document level, and would access it by `savedObject.attributes.attributes`

Reply:
Should we just call it `attrs`? Or is there something that makes sense to you? Aside from avoiding confusion, I don't have much of an opinion.

Reply:
I'm fine with `attributes`, just wanted to be sure. so `attributes.fields[name].displayName` would go in first, and others would follow (would existing fields be migrated later on?)

Reply:
sounds good to me!

Reply:
so, I've modified the PR to use the pattern we talked about, could you have a look on occasion? thx!